format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSource.java

```diff
@@ -193,19 +209,41 @@ public void prepareRun(BatchSourceContext context) throws Exception {
     }
 
     Path path = new Path(config.getPath(context));
-    FileSystem pathFileSystem = FileSystem.get(path.toUri(), conf);
+    FileSystem pathFileSystem = null;
+    try {
+      pathFileSystem = FileSystem.get(path.toUri(), conf);
+    } catch (IOException e) {
+      collector.addFailure("Failed to get FileSystem for path " + path, e.getMessage())
+        .withStacktrace(e.getStackTrace());
+      collector.getOrThrowException();
+    }
 
-    FileStatus[] fileStatus = pathFileSystem.globStatus(path);
+    FileStatus[] fileStatus = null;
+    try {
+      fileStatus = pathFileSystem.globStatus(path);
+    } catch (IOException e) {
+      collector.addFailure("Failed to get file status for path " + path, e.getMessage())
```

Comment:
use `String.format` everywhere.

`e.getMessage()`, is not a `correctiveAction`

also preserve error class name,

`%s: %s, e.getClass().getName(), e.getMessage()`